zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/user/CreateUserTest.java

```diff
@@ -99,27 +97,23 @@ public void shouldAddDefaultPermissionsOnUserCreation() {
     final var username = UUID.randomUUID().toString();
 
     // when
-    final var userKey =
-        ENGINE
-            .user()
-            .newUser(username)
-            .withName("Foo Bar")
-            .withEmail("foo@bar.com")
-            .withPassword("password")
-            .create()
-            .getKey();
+    ENGINE
+        .user()
+        .newUser(username)
+        .withName("Foo Bar")
+        .withEmail("foo@bar.com")
+        .withPassword("password")
+        .create();
 
     // then
     assertThat(
             RecordingExporter.authorizationRecords(AuthorizationIntent.CREATED)
-                .withOwnerKey(userKey)
+                .withOwnerId(username)
                 .getFirst()
                 .getValue())
-        .hasOwnerKey(userKey)
         .hasOwnerType(AuthorizationOwnerType.USER)
         .hasResourceType(AuthorizationResourceType.USER)
-        .hasOnlyPermissions(
-            new Permission().setPermissionType(PermissionType.READ).addResourceId(username),
-            new Permission().setPermissionType(PermissionType.UPDATE).addResourceId(username));
+        .hasResourceId(username)
```

Comment:
🔧 You should also verify that the authorizationKey is correctly set, and different from the userKey (see my comment above related to the test). If you do this, I think there is no need to create a proper unit test